affected-paths/core/src/test/kotlin/com/squareup/affected/paths/core/git/AffectedPathsTest.kt

```diff
@@ -0,0 +1,312 @@
+package com.squareup.affected.paths.core.git
+
+import com.squareup.affected.paths.core.CoreAnalyzer
+import com.squareup.affected.paths.core.CoreOptions
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.time.Duration.Companion.minutes
+
+class AffectedPathsTest {
+
+    @TempDir
+    lateinit var root: File
+
+    @Test
+    fun `Basic affected paths test`() = runTest(timeout = 3.minutes) {
+        // Prep
+        createSettingsFile(
+            rootDir = root,
+            contents = """
+                rootProject.name = 'blah'
+                include 'app'
+                include 'library'
+            """.trimIndent()
+        )
+
+        createModule(
+            rootDir = root,
+            name = "app",
+            contents =
+            """
+                plugins {
+                    id 'application'
+                }
+                
+                dependencies {
+                    implementation(':library')
+                }
+                """.trimIndent()
+        )
+
+        createModule(
+            rootDir = root,
+            name = "library",
+            contents =
+            """
+                plugins {
+                    id 'java'
+                }
+                """.trimIndent()
+        )
+
+        // Test
+        val analyzer = CoreAnalyzer(
+            CoreOptions(
+                directory = root.toPath(),
+                changedFiles = listOf("app/build.gradle")
+            )
+        )
+
+        val result = analyzer.analyze()
+
+        // Results
+        assertContentEquals(listOf("app", "library"), result.projectMap.keys)
+        assertContentEquals(listOf("app"), result.affectedResults.flatMap { it.affectedProjectPaths }.distinct())
+
+    }
+
+    @Test
+    fun `Included builds test`() = runTest(timeout = 3.minutes) {
+        // Prep
+        val build1 = File(root, "build1").apply { mkdirs() }
+        val build2 = File(build1, "build2").apply { mkdirs() }
```

Comment:
this is only a comment, but I wish we could all agree to use modern Java APIs instead of ancient Java APIs (`Path` instead of `File`).

Reply:
Yea, I have no excuse for this besides laziness. I'll make this change.